src/App.tsx

```diff
@@ -1,26 +1,291 @@
+    if (inputRef.current) {
+      inputRef.current.disabled = true;
+    }
+
+    setTempoTodo({
+      id: 0,
+      title: title,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    try {
+      const createdTodo = await addTodo(title);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+      setTempoTodo(null);
+
+      return true;
+    } catch {
+      setError('Unable to add a todo');
+      setTempoTodo(null);
+
+      return false;
+    } finally {
+      if (inputRef.current) {
+        inputRef.current.disabled = false;
+      }
+    }
+  };
+
+  const handleDeleteTodo = async (id: number) => {
+    setIdToLoad(prev => [...prev, id]);
+
+    try {
+      await deleteTodo(id);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+    } catch {
+      setError('Unable to delete a todo');
+    } finally {
+      setIdToLoad([]);
+      inputRef.current?.focus();
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setIdToLoad(completedTodos.map(todo => todo.id));
+
+    try {
+      await Promise.all(
+        completedTodos.map(async todo => {
+          try {
+            await deleteTodo(todo.id);
+            setTodos(prev => prev.filter(currTodo => todo.id !== currTodo.id));
+          } catch {
+            setError('Unable to delete a todo');
+          }
+        }),
+      );
+    } finally {
+      setIdToLoad([]);
+      inputRef.current?.focus();
+    }
+  };
+
+  const handleToggleTodo = async (todo: Todo) => {
+    setIdToLoad(prev => [...prev, todo.id]);
+
+    try {
+      const toggledTodo = await updateTodo(todo.id, {
+        completed: !todo.completed,
+      });
+
+      setTodos(currTodos =>
+        currTodos.map(currTodo => {
+          return currTodo.id === toggledTodo.id ? toggledTodo : currTodo;
+        }),
+      );
+    } catch {
+      setError('Unable to update a todo');
+    } finally {
+      setIdToLoad([]);
+    }
+  };
+
+  const handleToggleAll = async () => {
+    let todosToToggle = todos.filter(todo => !todo.completed);
+
+    if (todosToToggle.length === 0) {
+      setIdToLoad(todos.map(todo => todo.id));
+      todosToToggle = [...todos];
+    } else {
+      setIdToLoad(todosToToggle.map(todo => todo.id));
+    }
+
+    try {
+      await Promise.all(
+        todosToToggle.map(async todo => {
+          try {
+            const toggledTodo = await updateTodo(todo.id, {
+              completed: !todo.completed,
+            });
+
+            setTodos(prev =>
+              prev.map(currTodo => {
+                return currTodo.id === toggledTodo.id ? toggledTodo : currTodo;
+              }),
+            );
+          } catch {
+            setError('Unable to update a todo');
+          }
+        }),
+      );
+    } finally {
+      setIdToLoad([]);
+    }
+  };
+
+  const TodoTitleSubmit = async (todo: Todo) => {
```

Comment:
bad naming only classes have Uppercase
```suggestion
  const handleSubmit = async (todo: Todo) => {
```